### PR TITLE
[READY] Buyable spare mining shuttle

### PR DESCRIPTION
**Pull request by Dennok**
## About The Pull Request

Moved check_deploy to base map_template.

Add shuttlecapsule that can spawn empty mining shuttle shell.
Better implement this after #53993

image

Decals disappear until #54082 finished.

This shell has navigation and mining shuttle, console, so you can park shuttle on lavaland or near station. 
But shell don't has air from start.

Unpack restrictions same as for shelter capsules.

To prevent traffic jams this shuttle can fly only to self designed custom docking port.

I am so lazy to calculate and set the correct offsets for the shuttle docker console that I did #54274

## Why It's Good For The Game

More things to engineers to do.
Now, when your mining shuttle harassed by ashdrake or bubelgum, you can order new mining shuttle.

## Changelog
:cl:
add: Mining shuttlecapsule.  Now, when your mining shuttle harassed by ashdrake or bubelgum, you can order new mining shuttle.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
but you cant order a new mining shuttle because you cant get that capsule anywhere
also wont it cause immediate atmos changes from lavaland turfs and airless plating?

File: code/modules/mining/constructed_shuttles.dm [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+			if(SHELTER_DEPLOY_BAD_AREA)
+				src.loc.visible_message("<span class='warning'>\The [src] will not function in this area.</span>")
+			if(SHELTER_DEPLOY_BAD_TURFS, SHELTER_DEPLOY_ANCHORED_OBJECTS)
+				var/width = template.width
+				var/height = template.height
+				src.loc.visible_message("<span class='warning'>\The [src] doesn't have room to deploy! You need to clear a [width]x[height] area!</span>")
+
+		if(status != SHELTER_DEPLOY_ALLOWED)
+			used = FALSE
+			return*/
```
Comment (Ghilker) [inline code/modules/mining/constructed_shuttles.dm:47]: why all this section is commented out?

File: code/modules/mining/constructed_shuttles.dm [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+		used = FALSE
+
+		playsound(src, 'sound/effects/phasein.ogg', 100, TRUE)
+
+		//var/turf/T = deploy_location
```
Comment (Ghilker) [inline code/modules/mining/constructed_shuttles.dm:53]: another commented code

**Comment by Dennok** [comment]
> but you cant order a new mining shuttle because you cant get that capsule anywhere
> also wont it cause immediate atmos changes from lavaland turfs and airless plating?

Uhh i lost cargo ordering.
Airless plating airless only on initial place.

**Comment by Dennok** [comment]
Uhh, Sorry for some unintended merge problems.

File: _maps/shuttles/construction_mining.dmm [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+/obj/effect/turf_decal/stripes/line{
+	dir = 6
+	},
+/turf/open/floor/plating/airless,
+/area/shuttle/mining)
+"J" = (
+/obj/machinery/computer/camera_advanced/shuttle_docker{
+	jumpto_ports = list(mining_home = 1, mining_away = 1, landing_zone_dock = 1, mining_public = 1, construction_mining_custom = 1);
+	shuttleId = "construction_mining";
+	shuttlePortId = "construction_mining_custom"
```
Comment (Timberpoes) [inline _maps/shuttles/construction_mining.dmm:86]: I'm not allowing player-built shuttles with custom docking locations. This `construction_mining_custom` looks suspiciously like a player-defined docking location.

You're advertising this as a buyable spare mining shuttle, so it should be limited to all the same docking ports as the existing mining shuttle.

If this means you need to add a secondary docking areas to allow this to function properly, then that's fine.

Also this should not be a var edit hidden in a mapping file.

File: _maps/shuttles/construction_mining.dmm [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+/obj/effect/turf_decal/stripes/line{
+	dir = 1
+	},
+/turf/open/floor/plating/airless,
+/area/shuttle/mining)
+"c" = (
+/obj/machinery/computer/shuttle{
+	possible_destinations = "construction_mining_custom";
+	shuttleId = "construction_mining"
+	},
```
Comment (Timberpoes) [inline _maps/shuttles/construction_mining.dmm:18]: I'm not allowing player-built shuttles with custom docking locations. This `construction_mining_custom` looks suspiciously like a player-defined docking location.

You're advertising this as a buyable spare mining shuttle, so it should be limited to all the same docking ports as the existing mining shuttle.

If this means you need to add a secondary docking areas to allow this to function properly, then that's fine.

Also this should not be a var edit hidden in a mapping file.

**Review by Timberpoes — CHANGES_REQUESTED** [review]

**Pull request closed without merging**